pkg/action/executor/rescheduler.go

```diff
@@ -0,0 +1,261 @@
+			ParentObjectType:      parentRefObject.Kind,
+		}
+	}
+	targetObj := &turboaction.TargetObject{
+		TargetObjectUID:       string(originalPod.UID),
+		TargetObjectNamespace: originalPod.Namespace,
+		TargetObjectName:      originalPod.Name,
+		TargetObjectType:      turboaction.TypePod,
+	}
+	moveSpec := turboaction.MoveSpec{
+		Source:      originalPod.Spec.NodeName,
+		Destination: nodeIdentifier,
+	}
+	content := turboaction.NewTurboActionContentBuilder(turboaction.ActionMove, targetObj).
+		ActionSpec(moveSpec).
+		ParentObjectRef(parentObjRef).
+		Build()
+
+	// Build TurboAction.
+	action := turboaction.NewTurboActionBuilder(originalPod.Namespace, *actionItem.Uuid).
+		Content(content).
+		Create()
+	return &action, nil
+}
+
+func (r *ReScheduler) reSchedule(action *turboaction.TurboAction) (*turboaction.TurboAction, error) {
+	// 1. Setup consumer
+	actionContent := action.Content
+	// Move destination should not be empty
+	moveSpec, ok := actionContent.ActionSpec.(turboaction.MoveSpec)
+	if !ok || moveSpec.Destination == "" {
+		return nil, errors.New("Failed to setup re-scheduler as move destination is nil after pre-precess.")
+	}
+
+	var key string
+	if actionContent.ParentObjectRef.ParentObjectUID != "" {
+		key = actionContent.ParentObjectRef.ParentObjectUID
+	} else if actionContent.TargetObject.TargetObjectUID != "" {
+		key = actionContent.TargetObject.TargetObjectUID
+	} else {
+		return nil, errors.New("Failed to setup re-scheduler consumer: failed to retrieve the key.")
+	}
+	glog.V(3).Infof("The current re-scheduler consumer is listening on the key %s", key)
+	podConsumer := turbostore.NewPodConsumer(string(action.UID), key, r.broker)
+
+	// 2. Get the target Pod
+	targetObj := actionContent.TargetObject
+	originalPod, err := r.kubeClient.Pods(targetObj.TargetObjectNamespace).Get(targetObj.TargetObjectName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get the pod to be re-scheduled based on the given targetObject "+
+			"info: %++v", targetObj)
+	}
+	podIdentifier := util.BuildIdentifier(originalPod.Namespace, originalPod.Name)
+
+	// 3. delete pod
+	// TODO, we may want to specify a DeleteOption.
+	err = r.kubeClient.Pods(originalPod.Namespace).Delete(originalPod.Name, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to delete pod %s: %s", podIdentifier, err)
+	}
+	glog.V(3).Infof("Successfully delete pod %s.\n", podIdentifier)
+
+	// 4. create a pending pod if necessary
+	if actionContent.ParentObjectRef.ParentObjectUID == "" {
+		glog.V(2).Infof("Pod %s a standalone pod. Need to clone manually", podIdentifier)
+		// TODO need to wait the pod has been deleted complete.. A better way may related to DeleteOption?
+		time.Sleep(time.Second * 3)
+		podClone := &podClone{r.kubeClient}
+		err := podClone.clone(originalPod)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// 5. Wait for desired pending pod
+	// TODO, should we always block here?
+	p, ok := <-podConsumer.WaitPod()
```

Comment:
Same as before, we need to think a timeout mechanism to prevent thread hanging here.